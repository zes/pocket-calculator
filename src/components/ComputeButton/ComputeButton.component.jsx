import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './ComputeButton.css';

export default class ComputeButton extends PureComponent {
  onClick = () => {
    const { updateOperation, value } = this.props;
    updateOperation(value);
  };

  render() {
    const { className, value } = this.props;
    const buttonClassName = `button ${className}`;
    return (
      <button type="button" className={buttonClassName} onClick={this.onClick}>
        {value}
      </button>
    );
  }
}

ComputeButton.propTypes = {
  className: PropTypes.string.isRequired,
  updateOperation: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
