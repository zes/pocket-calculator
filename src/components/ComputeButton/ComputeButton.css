.button {
    background-color: #fff;
    display: inline-block;
    border: 0.1em solid #000;
    margin: 0 0.3em 0.3em 0;
    border-radius: 0.3em;
    box-sizing: border-box;
    font-weight: 600;
    color: #000;
    transition: all 0.3s;
    width: 50px;
    height: 50px;
}

button:hover {
    color:#fff;
    background-color:#000;
}

button:focus {
    outline:none;
}
