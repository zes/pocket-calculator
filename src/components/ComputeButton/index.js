export { default } from './ComputeButton.component';
