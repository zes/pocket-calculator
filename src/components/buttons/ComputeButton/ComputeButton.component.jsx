import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import GenericButton from '../GenericButton';

export default class ComputeButton extends PureComponent {
  render() {
    const { className, updateOperation, value } = this.props;
    return (
      <GenericButton
        action={() => updateOperation(value)}
        className={className}
        value={value}
      />
    );
  }
}

ComputeButton.propTypes = {
  className: PropTypes.string.isRequired,
  updateOperation: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};
