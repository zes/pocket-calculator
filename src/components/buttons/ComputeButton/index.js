export { default } from './ComputeButton.container';
