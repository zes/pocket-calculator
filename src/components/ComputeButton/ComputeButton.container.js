import { connect } from 'react-redux';
import { updateOperation } from '../../redux/actions';

import ComputeButton from './ComputeButton.component';

const mapDispatchToProps = {
  updateOperation,
};

export default connect(
  null,
  mapDispatchToProps,
)(ComputeButton);
