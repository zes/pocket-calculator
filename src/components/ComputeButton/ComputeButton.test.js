import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ComputeButton from './ComputeButton.component';

Enzyme.configure({ adapter: new Adapter() });

describe('[Component] <ComputeButton />', () => {
  const props = {
    value: '2',
    updateOperation: jest.fn(),
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<ComputeButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call updateOperation when user click on button', () => {
    const computeButton = shallow(<ComputeButton {...props} />);
    computeButton.instance().onClick();
    expect(props.updateOperation).toHaveBeenCalledWith(props.value);
  });
});
