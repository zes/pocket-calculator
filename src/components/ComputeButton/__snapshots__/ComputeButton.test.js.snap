// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <ComputeButton /> renders a snapshot 1`] = `
<button
  className="button x"
  onClick={[Function]}
  type="button"
>
  2
</button>
`;
